compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2090,6 +2090,6 @@ mod size_asserts {
     use super::*;
     // tidy-alphabetical-start
     static_assert_size!(ty::RegionKind<'_>, 24);
-    static_assert_size!(ty::TyKind<'_>, 24);
+    static_assert_size!(ty::TyKind<'_>, 32);
```

Comment:
I am personally not a big fan of supporting where-bounds inside of types and would like to only allow them in where-bounds. I also believe that this is where pretty much all of the perf regression is coming from.

I do expect that playing around with this feature is more annoying and ugly if we have to split binder for it, so 🤷 i guess i don't mind too much